tutorials/communication-protocols.md

```diff
@@ -0,0 +1,309 @@
+# Tessel 2 Module Communication Protocols
+Each of the two ports on Tessel 2 exposes ten pins for use in creating custom modules. Two of the pins are for power (3.3V and ground) and the other eight are settable GPIO (General Purpose Input and Output) pins that can be used to communicate with your module.
+
+All eight communication pins allow for communication through changes in voltage: some through variable voltage from 0V to 3.3V (analog pins), some through simple on/off (0V/3.3V) states, and some through special digital toggling referred to by their communication protocols. Communication protocols (such as SPI, I2C, and UART) involve a pre-defined set of pins used in a specific way. These protocols are used to encode complex messages. It's a lot like [Morse code](http://en.wikipedia.org/wiki/Morse_code), but for electronics.
+
+In embedded electronics, there are four common protocols and Tessel [supports them all in JavaScript](https://tessel.io/docs/hardwareAPI).
+
+  * [GPIO](#gpio)
+  * [SPI](#spi)
+  * [I2C](#i2c)
+  * [UART](#uart)
+
+This guide will provide a brief overview of the protocols and cover some of the strengths and weaknesses of each.
+
+You can see a mapping of which pins can be used for which protocols [on the Tessel docs page](https://github.com/tessel/t2-docs/blob/master/hardware-api.md#pin-mapping).
+
+## Quick Reference
+Most of the time, you will choose your protocol based on the parts you are using when designing your module. Other things to consider are the pins you have available, as well as your communication speed requirements. The following table can be used as a quick reference for the more detailed explanations of each protocol below.
+
+| Protocol 	|          # Pins Required              |          Max Speed            	|
+|:--------:	|:-------------------------------------:|:-------------------------------:|
+| GPIO     	|          **1**         	             	| 1kHz                            |
+| SPI      	| **3+** (MOSI, MISO, SCK + 1 GPIO pin) | 25MBit/s                        |
+| I2C      	|          **2** (SCL and SDA)        	| I2C0 - 1Mbit/s I2C1 - 400kbit/s	|
```

Comment:
I think the I2C value can be configured for either port on T2 and the speed can be either 100kHz or 400kHz (according to https://github.com/tessel/t2-firmware/pull/154/files#diff-84087272b70af9498872cf6f791003abR654).